nixos/modules/services/web-servers/nginx/vhost-options.nix

```diff
@@ -70,12 +66,24 @@ with lib;
       '';
     };
 
-    enableSSL = mkOption {
+    addSSL = mkOption {
       type = types.bool;
       default = false;
       description = "Whether to enable SSL (https) support.";
     };
 
+    onlySSL = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Whether to enable SSL (https) support and reject plain (http) connections.";
+    };
+
+    enableSSL = mkOption {
+      type = types.bool;
+      visible = false;
+      default = false;
+    };
+
     forceSSL = mkOption {
       type = types.bool;
       default = false;
```

Comment:
Description proposal:

    Whether to add a separate nginx server block that permanently redirects (301) all
    plain HTTP traffic to HTTPS. This option needs <literal>addSSL</literal> to be
    set to true.